sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -526,6 +516,113 @@ export class EventHubSender extends LinkEntity {
         }
 
         let waitTimer: any;
+
+        let onRejected: Func<EventContext, void>;
+        let onReleased: Func<EventContext, void>;
+        let onModified: Func<EventContext, void>;
+        let onAccepted: Func<EventContext, void>;
+        let onAborted: () => void;
+
+        onAborted = () => {
```

Comment:
Nit: this and the other `on*` functions assignments could be directly combined with their declarations above. Then you could convert the declarations from `let` to `const` as well.

Reply:
I believe there is a reason we cannot do this if we were to follow the ideal order of things like the below:

- set timer (this would need defining of `actionAfterTimeout` and `removeListeners`)
- init()
- attach listeners to sender
- send

The `removeListeners` would need to know which listeners to remove and therefore they are defined first and set later

@ramya0820 I would suggest to keep the changes to a minimum, so that a quick glance at the diff tells us exactly what was added/changed. Ideally in PRs I am all in favor of refactoring to make things better. But in this case, since we are all already confused on all the listeners being added/removed multiple promises etc, :) I am in favor of using the old order of things, and add the call to init() in the right place